### PR TITLE
cli templates.py u2p

### PR DESCRIPTION
**Pull request by tstrych**
unitetest to pytest conversion

I removed as much usage of `cli.factory` as possible. 
Please do not merge before @rplevka ack's this, he is owner of the ProvisioningTemplates.

`================== 11 passed, 6 warnings in 251.04s (0:04:11) ==================`

### CODE REVIEW

**Comment by tstrych** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_template.py

File: tests/foreman/cli/test_template.py [hunk 1]
```diff
@@ -14,232 +14,238 @@
...
+
+    :CaseLevel: Integration
+    """
+    new_template = make_template()
+    Template.add_operatingsystem(
+        {'id': new_template['id'], 'operatingsystem-id': module_os_with_minor.id}
+    )
+    new_template = Template.info({'id': new_template['id']})
+    os_string = '{} {}.{}'.format(
+        module_os_with_minor.name, module_os_with_minor.major, module_os_with_minor.minor
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_template.py:165]: f-strings?

File: tests/foreman/cli/test_template.py [hunk 2]
```diff
@@ -14,232 +14,238 @@
...
+    :CaseImportance: Medium
+
+    :BZ: 1395229
+    """
+    template = make_template()
+    Template.add_operatingsystem(
+        {'id': template['id'], 'operatingsystem-id': module_os_with_minor.id}
+    )
+    template = Template.info({'id': template['id']})
+    os_string = '{} {}.{}'.format(
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_template.py:189]: f-strings?

**Review by rplevka — APPROVED** [review]
Hi! thanks for help. It looks sane, just address @omkarkhatavkars comments, please.

**Review by lhellebr — APPROVED** [review]
LGTM

File: tests/foreman/cli/test_template.py (hunk 1 above)
Comment (tstrych) [reply tests/foreman/cli/test_template.py:165]: thanks :)

**Comment by tstrych** [comment]
code quality error not connected `tests/robottelo/test_func_shared.py:309: AssertionError`

**Pull request merged**